coil-base/src/main/java/coil/annotation/InternalCoilApi.kt

```diff
@@ -0,0 +1,15 @@
+package coil.annotation
+
+/**
+ * Marks declarations that are part of Coil's internal API. They should not be used outside of the `coil` package
+ * as their signatures and semantics will change between future releases without any warnings and without providing
+ * any migration aids.
+ */
+@MustBeDocumented
+@Retention(value = AnnotationRetention.BINARY)
+@RequiresOptIn(
+    level = RequiresOptIn.Level.ERROR,
+    message = "This is an internal Coil API that should not be used from outside of the `coil` package. " +
+        "No compatibility guarantees are provided."
+)
+annotation class InternalCoilApi
```

Comment:
This is necessary to avoid duplicating the interruption code in `coil-gif` and `coil-svg`. Maybe `withInterruptibleSource` can be public API one day, but probably doesn't make sense.